src/parser/mod.rs

```diff
@@ -103,6 +109,67 @@ pub struct KeyMetadata {
     pub size: u32,
 }
 
+/// A trait for parsing multiple types of values to memcached values.
+pub trait ToMemcachedValue {
+    /// Returns the length of the value in bytes.
+    fn length(&self) -> usize;
+    /// Writes the value to a writer.
+    fn write_to<W: AsyncWriteExt + Unpin>(
+        &self,
+        writer: &mut W,
+    ) -> impl Future<Output = Result<(), crate::Error>>;
```

Comment:
Could we collapse this into an `fn as_bytes(&self) -> &[u8]` method? (Or maybe returning `std::borrow::Cow<'_, [u8]>` if we need the option of returning owned values). This way we could avoid writing the uint types to the buffer twice and do so only once

Reply:
Thanks for the suggestion, I was actually trying to work out a way to do exactly that so I'll play around with it a bit more.